### PR TITLE
Added compatibility with features and evaulator class

### PR DESCRIPTION
**Pull request by pmohamma**
*Description of changes:*
Fixed the predictor and preprocessing files so that Rotbaum can now handle features as well, both static and dynamic. Additionally fixed those files to be compatible with the evaluator class in gluonts so that we can use backtesting.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/model/rotbaum/test_model.py [hunk 1]
```diff
@@ -17,53 +17,29 @@
...
-    for quantile in [0.1, 0.5, 0.9]:
-        predictions = list(predictor_instance.predict(data_for_pred))
-        preds = [t.quantile(quantile) for t in predictions]
-        record[f"quantile_loss_{quantile}"] = quantile_loss(
-            list(chain(*[ts["target"] for ts in data_true])),
-            list(chain(*preds)),
-            quantile,
-        )
+
+    evaluator = Evaluator(quantiles=[0.1, 0.5, 0.9], num_workers=0)
```
Comment (dcmaddix) [inline test/model/rotbaum/test_model.py]: Are you able to call `backtest_metrics` directly?

File: test/model/rotbaum/test_model.py [hunk 2]
```diff
@@ -17,53 +17,29 @@
...
+    record, item_metrics = evaluator(iter(ts_it), iter(forecast_it), num_series=len(dsinfo.test_ds))
+
     if dsinfo["name"] == "constant":
         for q in [0.1, 0.5, 0.9]:
-            assert record[f"quantile_loss_{q}"] == 0
+            print(record[f"wQuantileLoss[{q}]"])
```
Comment (dcmaddix) [inline test/model/rotbaum/test_model.py]: Remove `print`

File: test/model/rotbaum/test_model.py [hunk 3]
```diff
@@ -17,53 +17,29 @@
...
+            assert record[f"wQuantileLoss[{q}]"] == 0
     if dsinfo["name"] == "synthetic":
-        assert 1.1 < record[f"quantile_loss_{0.5}"] < 1.2
+        print(record[f"wQuantileLoss[0.5]"])
```
Comment (dcmaddix) [inline test/model/rotbaum/test_model.py]: Remove `print`

File: test/model/rotbaum/test_model.py [hunk 4]
```diff
@@ -17,53 +17,29 @@
...
-        predictions = list(predictor_instance.predict(data_for_pred))
-        preds = [t.quantile(quantile) for t in predictions]
-        record[f"quantile_loss_{quantile}"] = quantile_loss(
-            list(chain(*[ts["target"] for ts in data_true])),
-            list(chain(*preds)),
-            quantile,
-        )
+
+    evaluator = Evaluator(quantiles=[0.1, 0.5, 0.9], num_workers=0)
+    record, item_metrics = evaluator(iter(ts_it), iter(forecast_it), num_series=len(dsinfo.test_ds))
```
Comment (dcmaddix) [inline test/model/rotbaum/test_model.py]: Can we change from `record` to `agg_metrics`?

File: test/model/rotbaum/test_model.py (hunk 3 above)
Comment (dcmaddix) [inline test/model/rotbaum/test_model.py]: The `test_accuracy` in https://github.com/awslabs/gluon-ts/blob/master/test/model/conftest.py#L106 used for all models also checks ` assert agg_metrics["ND"] <= accuracy` so we can add this as well, where `accuracy = 10.0` in the synthetic dataset case.

File: src/gluonts/model/rotbaum/_predictor.py [hunk 5]
```diff
@@ -134,10 +143,9 @@ def __init__(
     @validated()
     def __call__(self, training_data):
         assert training_data
-        if self.freq is not None:
+        assert self.freq is not None:
+        if next(iter(training_data))["start"].freq is not None:
```
Comment (dcmaddix) [inline src/gluonts/model/rotbaum/_predictor.py:172]: Nice fix here, thanks!

**Review by dcmaddix — APPROVED** [review]

File: test/model/rotbaum/test_model.py [hunk 6]
```diff
@@ -17,53 +17,28 @@
...
-        for ts in dsinfo.test_ds
-    ]
-    for quantile in [0.1, 0.5, 0.9]:
-        predictions = list(predictor_instance.predict(data_for_pred))
-        preds = [t.quantile(quantile) for t in predictions]
-        record[f"quantile_loss_{quantile}"] = quantile_loss(
-            list(chain(*[ts["target"] for ts in data_true])),
-            list(chain(*preds)),
-            quantile,
+    predictor(dsinfo.train_ds)
```
Comment (dcmaddix) [inline test/model/rotbaum/test_model.py]: We can add a `TODO` for a `train` method so it matches the `predictor = estimator.train` library structure.

**Review by dcmaddix — APPROVED** [review]

File: src/gluonts/model/rotbaum/_estimator.py [hunk 7]
```diff
@@ -38,11 +42,28 @@ class ThirdPartyEstimator(Estimator):
     def __init__(self, predictor_cls: type, **kwargs) -> None:
         self.predictor = predictor_cls(**kwargs)
 
+    @override()
```
Comment (dcmaddix) [inline src/gluonts/model/rotbaum/_estimator.py]: What does the @override() do?

File: src/gluonts/model/rotbaum/_estimator.py [hunk 8]
```diff
@@ -38,11 +42,28 @@ class ThirdPartyEstimator(Estimator):
...
     ) -> Predictor:
         return self.predictor(training_data)
 
+    def train_model(self):
+        return TrainOutput(
+            transformation=self.create_transformation(),
+            trained_net=self.create_training_network(),
+            predictor=self.create_predictor(),
+        )
+
```
Comment (dcmaddix) [inline src/gluonts/model/rotbaum/_estimator.py:56]: Looks great thanks for adding!

File: src/gluonts/model/rotbaum/_estimator.py [hunk 9]
```diff
@@ -38,11 +42,28 @@ class ThirdPartyEstimator(Estimator):
...
+    def create_predictor(self) -> Predictor:
+        return self.predictor
+
+    def create_transformation(self) -> Transformation:
```
Comment (dcmaddix) [inline src/gluonts/model/rotbaum/_estimator.py:60]: Would we want to add any transformations here like in the other models with the time and age features?

**Review by dcmaddix — APPROVED** [review]

File: src/gluonts/model/rotbaum/_estimator.py (hunk 9 above)
Comment (pmohamma) [reply src/gluonts/model/rotbaum/_estimator.py:60]: I was unsure if we wanted that, but when looking into it, it looked like transformations were passed into training networks which would require us to redo the architecture of Rotbaum. I'm not sure if there is another way to work with transformations, but can discuss this offline.

**Pull request merged**